ts/src/provider.ts

```diff
@@ -9,15 +9,25 @@ import {
   SimulatedTransactionResponse,
   Commitment,
   SendTransactionError,
+  SendOptions,
 } from "@solana/web3.js";
 import { bs58 } from "./utils/bytes/index.js";
 import { isBrowser } from "./utils/common.js";
 
+export interface Provider {
+  readonly connection: Connection,
+
+  send?(req: SendTxRequest, opts?: SendOptions): Promise<TransactionSignature>;
```

Comment:
question: should we use `SendTxRequest` or should we upack it `tx: Transaction, signers?: Array<Signer>`

Reply:
also, why is `SendOptions` not in `SendTxRequest`? Seems like we should everything together or none of them

Reply:
Let's unpack it all to be consistent with the existing apis.